### PR TITLE
Plugin fixes and improvements

### LINKED ISSUE #7507: Rotating plugin search while empty closes keyboard
> If I tap to start a new search (but don’t actually type anything yet) and then rotate my device, the search box and keyboard close and I’m left on a blank “Search Plugins” screen. If I then search for a plugin, then go back, I’m taken back to that blank screen before going back to the main plugins screen.

### PR DESCRIPTION
**Pull request by oguzkocer**
Fixes #7507, #7400, #7367. This PR has several fixes and improvements to Plugins. The commit messages are (hopefully) self-documenting. Since this PR is a mix of different improvements, I strongly suggest reviewing the PR commit by commit as they are hopefully well-contained.

To test:
* Test the issue described in #7507.
* Check that Akismet, VaultPress and Jetpack are marked as `auto-managed` for your Automated Transfer site
* Check that all actions (install, update, delete) are disabled for auto-managed plugins in plugin detail page
* Check that Jetpack (or Akismet) is not labeled as `auto-managed` for a regular site (non-AT)
* General testing of plugin actions (install, update, delete etc) - just a monkey test should be fine.

If you don't have an Automated Transfer site, ping me on Slack and I can invite you to one. Let me know if you have any questions!

screenshot_1528148742
screenshot_1528148747

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginUtils.java [hunk 1]
```diff
@@ -46,4 +54,27 @@ static boolean isUpdateAvailable(@Nullable ImmutablePluginModel immutablePlugin)
             return !installedVersionStr.equalsIgnoreCase(availableVersionStr);
         }
     }
+
+    public static boolean isJetpack(@NonNull ImmutablePluginModel plugin) {
+        return checkNameOfPlugin(plugin, "jetpack/jetpack");
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginUtils.java]: Can we use `JETPACK_PLUGIN_NAME` here?

File: WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginUtils.java [hunk 2]
```diff
@@ -46,4 +54,27 @@ static boolean isUpdateAvailable(@Nullable ImmutablePluginModel immutablePlugin)
...
+            return false;
+        }
+        boolean isAutoManaged = false;
+        for (String pluginName : AUTO_MANAGED_PLUGINS) {
+            isAutoManaged = isAutoManaged || checkNameOfPlugin(plugin, pluginName);
+        }
+        return isAutoManaged;
+    }
+
+    private static boolean checkNameOfPlugin(@NonNull ImmutablePluginModel plugin, @NonNull String name) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginUtils.java]: Is this method necessary? Seems like `StringUtils.equals(plugin.getName(), name)` would suffice, but if we do keep it a better name would be helpful (`isSamePluginName` perhaps?).

**Comment by oguzkocer** [comment]
@nbradbury Thanks for the review. Fixed both issues in a440817c292238eebc613ffa57ff21eab2616bdd. Ready for a second pass.

**Comment by nbradbury** [comment]
:shipit:

**Pull request merged**